### DISCUSSION TITLE
ABOUT MODEL PREDECTION

### DISCUSSION THREAD

**Discussion by yvr354** [ultralytics/ultralytics#9552] [General]
I am working on a bread toast project and trying to predict broken bread toast. The model works well. However, I ran into an issue when I found predated images.  I have several similar photos in my test dataset. For one image, it shows 64%, but for the duplicated image, the model does not predict anything. I use yolov8-ghost-p2.yaml and yolov8s.pt for training.
  Can someone help me with this problem?

![2620](https://github.com/ultralytics/ultralytics/assets/68755664/f274d15b-3cfb-4cc7-9e9f-e3d8d46b8a09)
![2634](https://github.com/ultralytics/ultralytics/assets/68755664/d1199951-1953-4aeb-be32-a94713f5c373)

  .

**Comment by pderrenger** [comment]
Hello! It's great to hear that the model works well for your bread toast project. 🍞 However, the inconsistency in prediction between duplicated images is indeed intriguing. A possible reason for this could include slight differences in image encoding, compression artifacts, or even minor variations unnoticed by the human eye that can affect model prediction.

It may help to try the following approach to ensure consistency:
- Preprocess your images uniformly before running predictions. This could include steps like resizing, normalization, etc., as per your training setup.
- Experiment with different confidence thresholds. Sometimes, predictions might be borderline and altering the threshold slightly could yield consistent results.

For example, to predict with a different confidence threshold, you can adjust the `conf` parameter:
```python
from ultralytics import YOLO

# Load the model
model = YOLO('yolov8s.pt')

# Predict with a custom confidence threshold
results = model.predict('path/to/image.jpg', conf=0.65)
```

These differences highlight the importance of preprocessing and parameter tuning for achieving consistent results across similar images. Hope this helps! Feel free to share if there are any specific preprocessing steps you've taken or any further insights into the issue. 🛠️
